chainerx_cc/chainerx/python/routines.cc

```diff
@@ -371,6 +371,106 @@ void InitChainerxLogic(pybind11::module& m) {
+      }
+      // numpy.ndarray
+      if (py::isinstance<py::array>(indices_or_sections)) {
+          py::array np_ios = py::cast<py::array>(indices_or_sections);
+          if (np_ios.ndim() >= 2) {
+              throw py::value_error{std::string{"Too many dimensions of indices: "} + std::to_string(np_ios.ndim())};
+          }
+          // sections: scalar
+          if (np_ios.ndim() == 0) {
+              int64_t sections{};
+              py::object scalar_np = np_ios.attr("tolist")();
+              if (py::isinstance<py::int_>(scalar_np)) {
+                  sections = py::cast<int64_t>(scalar_np);
+              } else if (py::isinstance<py::float_>(scalar_np)) {
+                  sections = pyfloat_to_sections_or_value_error(scalar_np);
+              } else {
+                  throw py::type_error{"Sections must be an integer."};
+              }
+              return split_sections(ary, sections, axis);
+          }
+
+           // indices: (0,)-shape
+          if (np_ios.size() == 0) {
+              return split_indices(ary, {}, axis);
+          }
+
+           if (np_ios.dtype().kind() != 'i') {
+              throw py::type_error{std::string{"Indices must be integers."}};
+          }
+          // indices: non-scalar
+          std::vector<int64_t> indices{};
+          py::list indices_pylist = np_ios.attr("tolist")();
+          for (py::handle item : indices_pylist) {
+              indices.emplace_back(py::cast<int64_t>(item));
+          }
+
+           return split_indices(ary, indices, axis);
+      }
+      // indices: sequence
+      if (py::isinstance<py::sequence>(indices_or_sections)) {
+          std::vector<int64_t> indices{};
+          try {
+              indices = py::cast<std::vector<int64_t>>(indices_or_sections);
+          } catch (const py::cast_error& e) {
+              throw py::type_error{std::string{"Indices not understood: "} + py::cast<std::string>(py::repr(indices_or_sections))};
+          }
+
+                   return split_indices(ary, indices, axis);
+              }
+              throw py::type_error{std::string{"indices_or_sections not understood: "} +
+                                   py::cast<std::string>(py::repr(indices_or_sections))};
+}
+
+ auto split(const ArrayBodyPtr& ary, py::handle indices_or_sections, int8_t axis) {
+      auto split_sections = [](const ArrayBodyPtr& ary, int64_t sections, int8_t axis) {
+            return MoveArrayBodies(Split(Array{ary}, sections, axis));
+      };
+      auto split_indices = [](const ArrayBodyPtr& ary, const std::vector<int64_t>& indices, int8_t axis) {
+            return MoveArrayBodies(Split(Array{ary}, indices, axis));
+      };
+      return _split(split_sections, split_indices, ary, indices_or_sections, axis);
+}
+
+ auto dsplit(const ArrayBodyPtr& ary, py::handle indices_or_sections) {
+      auto split_sections = [](const ArrayBodyPtr& ary, int64_t sections, int8_t axis) {
```

Comment:
Could you comment out `axis` to avoid `variable set but unused error`?
```suggestion
      auto split_sections = [](const ArrayBodyPtr& ary, int64_t sections, int8_t /* axis */) {
```